src/CMakeLists.txt

```diff
@@ -0,0 +1,82 @@
+set(SRC glacierapp.cpp)
+set(HEADERS glacierapp.h)
+
+add_library(glacierapp SHARED ${SRC})
```

Comment:
You didn't add the HEADERS to the library source files thus AUTOMOC won't see them and won't generate the appropriate code. This issue does not break the build because the `GlacierApp` class does not declare its own Qt signals, slots or enums, however, the meta-information about the class name will be lost (and thus will affect debug output).
Fix:
```add_library(glacierapp SHARED ${SRC} ${HEADERS})```